pkg/pkg.bzl

```diff
@@ -379,3 +379,64 @@ def pkg_deb(name, package, **kwargs):
         changes = out_changes,
         **kwargs
     )
+
+def _format_zip_file_arg(f):
+    return "%s=%s" % (_quote(f.path), dest_path(f, strip_prefix = None))
+
+def _pkg_zip_impl(ctx):
+    args = ctx.actions.args()
+
+    args.add("-o", ctx.outputs.out.path)
+    args.add("-d", ctx.attr.package_dir)
+    args.add("-t", ctx.attr.timestamp)
+
+    args.add_all(
+        ctx.files.srcs,
+        map_each = _format_zip_file_arg,
+    )
+
+    args.set_param_file_format("multiline")
+    args.use_param_file("@%s")
```

Comment:
I manually tested this (on Linux/Mac/Windows) by setting `use_always = True` and running the tests...
Do you want to see a test for this? Maybe we could generate a really big zip file. Or maybe use this undocumented option: https://github.com/bazelbuild/bazel/blob/3a879281507497dfbad1dda0051368a36f6e0e2b/src/main/java/com/google/devtools/build/lib/analysis/config/CoreOptions.java#L119

Reply:
We can add that later.